packages/rulesets/src/arazzo/index.ts

```diff
@@ -0,0 +1,258 @@
+import arazzoStepSuccessActionsValidation from './functions/arazzoStepSuccessActionsValidation';
+import arazzoWorkflowDependsOnValidation from './functions/arazzoWorkflowDependsOnValidation';
+import arazzoStepSuccessCriteriaValidation from './functions/arazzoStepSuccessCriteriaValidation';
+import arazzoStepRequestBodyValidation from './functions/arazzoStepRequestBodyValidation';
+import arazzoStepValidation from './functions/arazzoStepValidation';
+
+export default {
+  documentationUrl: 'https://meta.stoplight.io/docs/spectral/docs/reference/arazzo-rules.md',
+  formats: [arazzo1_0],
+  rules: {
+    'arazzo-document-schema': {
+      description: 'Arazzo Document must be valid against the Arazzo schema.',
+      message: '{{error}}',
+      recommended: true,
+      severity: 0,
+      given: '$',
+      then: {
+        function: arazzoDocumentSchema,
+      },
+    },
+    'arazzo-workflowId-unique': {
+      description: 'Every workflow must have unique "workflowId".',
+      recommended: true,
+      severity: 0,
+      given: '$',
+      then: {
+        function: arazzoWorkflowIdUniqueness,
+      },
+    },
+    'arazzo-workflow-output-validation': {
+      description: 'Every workflow output must have unique name and its value must be a valid runtime expression.',
+      message: `{{error}}`,
+      recommended: true,
+      severity: 0,
+      given: '$',
+      then: {
+        function: arazzoWorkflowOutputNamesValidation,
+      },
+    },
+    'arazzo-workflow-stepId-unique': {
+      description: 'Every step must have unique "stepId".',
+      message: `{{error}}`,
+      recommended: true,
+      severity: 0,
+      given: '$.workflows[*]',
+      then: {
+        function: arazzoStepIdUniqueness,
+      },
+    },
+    'arazzo-step-output-validation': {
+      description: 'Every step output must have unique name and its value must be a valid runtime expression.',
+      message: `{{error}}`,
+      recommended: true,
+      severity: 0,
+      given: '$',
+      then: {
+        function: arazzoStepOutputNamesValidation,
+      },
+    },
+    'arazzo-step-parameters-validation': {
+      description: 'Step parameters and workflow parameters must valid.',
+      message: `{{error}}`,
+      recommended: true,
+      severity: 0,
+      given: '$',
+      then: {
+        function: arazzoStepParametersValidation,
+      },
+    },
+    'arazzo-step-failure-actions-validation': {
+      description:
+        'Every failure action must have a unique "name", and the fields "workflowId" and "stepId" are mutually exclusive.',
+      message: `{{error}}`,
+      recommended: true,
+      severity: 0,
+      given: '$',
+      then: {
+        function: arazzoStepFailureActionsValidation,
+      },
+    },
+    'arazzo-step-success-actions-validation': {
+      description:
+        'Every success action must have a unique "name", and the fields "workflowId" and "stepId" are mutually exclusive.',
```

Comment:
These could be two different rules IMO but fine like this too. The second part could be done with XOR function.

Reply:
It's all wrapped up in the custom function for now at least. Some of it is captured by the JSON Schema check too